src/py/flwr/server/strategy/dp_strategy_wrapper_fixed_clipping.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Central differential privacy with fixed clipping.
+
+Papers: https://arxiv.org/abs/1712.07557, https://arxiv.org/abs/1710.06963
+"""
+
+
+from logging import WARNING
+from typing import Dict, List, Optional, Tuple, Union
+
+from flwr.common import (
+    EvaluateIns,
+    EvaluateRes,
+    FitIns,
+    FitRes,
+    NDArrays,
+    Parameters,
+    Scalar,
+    ndarrays_to_parameters,
+    parameters_to_ndarrays,
+)
+from flwr.common.differential_privacy import (
+    add_gaussian_noise_to_params,
+    compute_clip_model_update,
+)
+from flwr.common.differential_privacy_constants import CLIENTS_DISCREPANCY_WARNING
+from flwr.common.logger import log
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy.strategy import Strategy
+
+
+class DPStrategyWrapperServerSideFixedClipping(Strategy):
+    """Wrapper for Central DP with Server Side Fixed Clipping.
+
+    Parameters
+    ----------
+    strategy : Strategy
+        The strategy to which DP functionalities will be added by this wrapper.
+    noise_multiplier : float
+        The noise multiplier for the Gaussian mechanism for model updates.
+        A value of 1.0 or higher is recommended for strong privacy.
+    clipping_norm : float
+        The value of the clipping norm.
+    num_sampled_clients : int
+        The number of clients that are sampled on each round.
+
+    Examples
+    --------
+    Create an strategy:
+
+    >>> strategy = fl.server.strategy.FedAvg( ... )
+
+    Wrap the strategy with a DP wrapper
+
+    >>> dpStrategy = DPStrategyWrapperServerSideFixedClipping(
+                        strategy,
+                        cfg.noise_multiplier,
+                        cfg.clipping_norm,
+                        cfg.num_sampled_clients
+                        )
```

Comment:
Code examples need `>>>` for every line and they should follow the same formatting as other code